lib/timescaledb/models/model.ex

```diff
@@ -3,10 +3,43 @@ defmodule Membrane.Telemetry.TimescaleDB.Model do
   Module responsible for putting data to TimescaleDB.
   """
 
-  require Logger
-  import Ecto.Query
   alias Membrane.Telemetry.TimescaleDB.Repo
-  alias Membrane.Telemetry.TimescaleDB.Model.{Measurement, ComponentPath, Link}
+  alias Membrane.Telemetry.TimescaleDB.Model.{Element, Measurement, ComponentPath, Link}
+
+  import Ecto.Query
+
+  require Logger
+
+  def add_all_measurements(measurements) do
+    component_paths =
+      measurements
+      |> Enum.map(&%{path: &1.component_path})
+      |> Enum.uniq()
+
+    try do
+      Ecto.Multi.new()
+      |> insert_all_component_paths(component_paths)
+      |> fetch_remaining_paths(component_paths)
+      |> insert_all_measurements(measurements)
+      |> Repo.transaction()
+    rescue
+      error in Postgrex.Error -> {:error, error}
+    end
+  end
+
+  def add_measurement(measurement) do
+    Measurement.changeset(%Measurement{}, measurement) |> Repo.insert()
+  end
+
+  def add_link(link) do
+    Link.changeset(%Link{}, link) |> Repo.insert()
+  end
```

Comment:
In these functions we can also add additional pipe operators
```suggestion
  def add_measurement(measurement) do
    %Measurement{}
    |> Measurement.changeset(measurement) 
    |> Repo.insert()
  end

  def add_link(link) do
    %Link{}
    |> Link.changeset(link) 
    |> Repo.insert()
  end
```